content/blog/2020-08-14-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,242 @@
+identities <- read_excel("data/superheroes.xlsx", sheet = "alter_egos")
+```
+
+Let's take a look at the sheet with data on superheroes.
+
+```{r}
+superheroes
+```
+
+And now let's take a look at the sheet listing the alter ego of each character.
+
+```{r}
+identities
+```
+
+In each sheet, there is a `hero` column that allows us to match data from one sheet to the other.   
+Using `left_join()` from the dplyr package, we can bring in the data from the identities data frame into our superheroes data frame.
+
+To do this we'll call `left_join()` on superheroes, putting the identities data set as our first argument and specify our common variable with the `by` argument.
+
+```{r}
+superhero_identities <- superheroes %>%
+  left_join(identities, by = "hero")
+
+superhero_identities
+```
+
+Great, we've joined the data frames with no problem!  For parity, let's also walk through the spreadsheet approach and utilize a VLOOKUP to bring together these data.
+
+![VLOOKUP](screenshots/VLOOKUP.png)
+
+There are, of course, instances in which we'll need to pull in data from multiple sheets, and the superheroes data set is no exception. 
+We've been informed there are an additional two sheets that have recently been added. One containing information on the main power or skill of each character.
+
+```{r}
+powers <- read_excel("data/superheroes.xlsx", sheet = "powers")
+powers
+```
+
+And the other containing the occupation of the alter-ego of each superhero. 
+
+```{r}
+occupation <- read_excel("data/superheroes.xlsx", sheet = "occupation")
+occupation
+```
+
+We see that the powers sheet has a hero column allowing for another simple left join. However, the occupation sheet doesn't have a common column by that specific name, but we do recognize that the information in the name column is the same for the alter-ego column from our identities spreadsheet.
+We can take advantage of this relationship to bring these data frames together as shown in the code below. 
+
+```{r}
+heroes_full <- superhero_identities %>%
+  left_join(powers, by = "hero") %>%
+  left_join(occupation, by = c("alter_ego" = "name"))
+```
+
+Here we conduct a simple left join bringing in the data from the powers data set, and then another left join to bring in the data from the occupation spreadsheet.
+In the `by` argument we specify that the column `alter_ego` should be matched to the `name` column from the occupation data frame. 
+
+We've expanded our dataset quite a bit. Let's take an alternative approach to view it using the `glimpse()` function from dplyr. 
+
+```{r}
+glimpse(heroes_full)
+```
+
+For doing this same task in Excel, we can continue to utilize VLOOKUPS for each additional column of data we want to pull into our main sheet.  
+To bring in our occupation data we simply adjust the initial reference point in our formula from column A2 which houses hero name to E2 which houses alter egos which was also added via a VLOOKUP. 
```

Comment:
This example turned out well, really demonstrating the pattern of pulling data spread out over various sheets into one object.